enhancements/sig-architecture/34-work-executor-group/README.md

```diff
@@ -0,0 +1,253 @@
+
+Currently all the API requests raised by work agent in the managed cluster are
+from a fixed service account ("open-cluster-management-agent/klusterlet-work-sa"
+under the default installation). However in some cases, we need to clarify the
+source/owner identity of the `ManifestWork` resource so that we can record the
+api audits in a finer granularity. This will be helpful for the managed
+cluster's admin to be capable of knowing "who is operating that `ManifestWork`
+from the hub cluster".
+
+### Goals
+
+- `ManifestWork` resource be able to clarify the executor/owner identity before
+  actually writing the resources to the managed cluster.
+
+### Non-Goals
+
+- This proposal doesn't put any requirement to the selection of executor/owner in
+  the "ManifestWork", which allows hub admin to reference any valid Kubernetes 
+  user/role to be the executor/owner.
+  
+## Terms
+
+- `Executor Subject`: The subject which is used by the work agent when actually 
+  applying resources in the spoke cluster.
+
+
+## Function Spec
+
+1. A hub admin can limit the scope of the dispatch'able resources for a 
+   `ManifestWork` by setting a non-empty executor subject, so the work agent
+   in the spoke cluster can report failure when applying/modifying out-of-scope 
+   resources.
+2. A spoke admin can expand the permission scope of the executor subject 
+   (e.g. a service-account) by binding more ClusterRole/Role to it.
+3. (beta phase) A hub user should pass the authorization before writing
+   the resources in the work object. The hub cluster holds a policy configuration
+   of "which hub user has write access to which work resource" and a validating
+   admission webhook will handle the execution of that policy.
+
+## Proposal
+
+### API Changes
+
+This proposal adds a new field `executorSubject` to the `ManifestWork` api, 
+which is defaulted to nil if absent. A nil executor subject indicates that the 
+work agent doesn't perform any impersonation before requesting the hosting 
+managed cluster.
+
+
+A sample of the new `executorSubject` field in Yaml is shown below:
+
+```yaml
+apiVersion: work.open-cluster-management.io/v1
+kind: ManifestWork
+metadata: ...
+spec:
+  executorSubject:
+    type: [ServiceAccount] # union discriminator for extensibility
+    serviceAccount:
+      name: ...
+      namespace: ...
+  workload:
+    manifests: [...]
+```
+
+### Work-agent client permission muzzling
+
+The supported executor subject types will be merely "ServiceAccount".
+For "ServiceAccount" type, intuitively the referenced service account is 
+working like an "ambassador" identity, and the work agent will need to 
+explicitly check whether that "ambassador" has sufficient RBAC permission 
+in the managed cluster via SubjectAccessReview api. Due to the limit of 
+the SubjectAccessReview api, we need to send (# of contained objects) times 
+SAR requests to the spoke cluster, but this should be fine b/c the work-agent 
```

Comment:
Yes, and you'll be able to cache a result if you like.

Reply:
added that in the new commit